### PR TITLE
feat: Upgrade core-contracts lib to use redesigned device profile

### LINKED ISSUE #815: Upgrade core-contracts lib to use redesigned device profile
Upgrade core-contracts lib to apply the v2 device profile changes:
- Upgrade core-contracts lib to use the redesigned device profile
- Update swagger file

### PR DESCRIPTION
**Pull request by weichou1229**
Close #815 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/master/.github/CONTRIBUTING.md

## What is the new behavior?
Upgrade core-contracts lib to apply the v2 device profile changes:
- Upgrade core-contracts lib to use the redesigned device profile
- Update swagger file

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by lenny-intel — DISMISSED** [review]
LGTM

File: example/cmd/device-simple/res/Simple-Driver.yaml [hunk 1]
```diff
@@ -49,35 +52,18 @@ deviceResources:
 deviceCommands:
```
Comment (cloudxxx8) [inline example/cmd/device-simple/res/Simple-Driver.yaml:52]: You miss the `isHidden: false` in deviceCommands.
This is an example, so we have to demonstrate the sample clearly although the default value is false

File: example/cmd/device-simple/res/Simple-Driver.yaml [hunk 2]
```diff
@@ -49,35 +52,18 @@ deviceResources:
...
   -
     name: "Switch"
-    get:
-      - { deviceResource: "SwitchButton" }
-    set:
-      - { deviceResource: "SwitchButton", parameter: "false" }
+    readWrite: "W"
```
Comment (cloudxxx8) [inline example/cmd/device-simple/res/Simple-Driver.yaml:55]: The original sample is `RW`

File: example/cmd/device-simple/res/configuration.toml [hunk 3]
```diff
@@ -81,7 +81,7 @@ RetryWaitPeriod = "1s"
   [[DeviceList.AutoEvents]]
     Frequency = '10s'
     OnChange = false
-    SourceName = 'Switch'
+    SourceName = 'SwitchButton'
```
Comment (cloudxxx8) [inline example/cmd/device-simple/res/configuration.toml:84]: You don't need to make this change if it is RW

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: openapi/v2/device-sdk.yaml [hunk 4]
```diff
@@ -173,48 +145,43 @@ components:
...
       type: object
       properties:
-        created:
-          type: integer
-          description: "A Unix timestamp indicating when the object was initially persisted to a database"
-        modified:
-          type: integer
-          description: "A Unix timestamp indicating when the object was last modified"
-        description:
+        apiVersion:
```
Comment (cloudxxx8) [inline openapi/v2/device-sdk.yaml:148]: apiVersion will only be in request, response, and event

File: example/cmd/device-simple/res/Simple-Driver.yaml (hunk 1 above)
Comment (weichou1229) [reply example/cmd/device-simple/res/Simple-Driver.yaml:55]: Added the isHidden fields.

File: example/cmd/device-simple/res/Simple-Driver.yaml (hunk 2 above)
Comment (weichou1229) [reply example/cmd/device-simple/res/Simple-Driver.yaml]: Modified.

File: example/cmd/device-simple/res/configuration.toml (hunk 3 above)
Comment (weichou1229) [reply example/cmd/device-simple/res/configuration.toml]: Modified.

File: openapi/v2/device-sdk.yaml (hunk 4 above)
Comment (weichou1229) [reply openapi/v2/device-sdk.yaml]: Removed the `apiVersion ` from the device profile schema.

**Comment by cloudxxx8** [comment]
@weichou1229  please help resolve the conflict

**Comment by weichou1229** [comment]
Rebased and resolved the conflict.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**